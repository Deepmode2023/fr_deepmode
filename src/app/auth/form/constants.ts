import { FormInputsType } from "@/components/Form/Form";
export const BreadcrompsItem = [
  { title: "login", value: "LOGIN" },
  { title: "registretion", value: "REGESTRETION" },
];

export const FormInputsConstant = (formKey: string): FormInputsType[] => {
  switch (formKey) {
    case "REGESTRETION":
      return [
        {
          label: "Surname",
          name: "surname",
          placeholder: "Enter your surname",
          type: "text",
          classname: {
            container: "col-span-1",
            inputBasic: {
              input: "bg-[transparent] text-white",
              inputContainer: "border-2 p-[10px] rounded-md",
            },
          },
        },
        {
          label: "Name",
          name: "name",
          placeholder: "Enter your name",
          type: "text",
          classname: {
            container: "col-span-1",
            inputBasic: {
              input: "bg-[transparent] text-white",
              inputContainer: "border-2 p-[10px] rounded-md",
            },
          },
        },
        {
          label: "Email",
          name: "email",
          placeholder: "Enter your email",
          type: "email",
          classname: {
            container: "col-span-2",
            inputBasic: {
              input: "bg-[transparent] text-white",
              inputContainer: "border-2 p-[10px] rounded-md",
            },
          },
        },
        {
          label: "password",
          name: "password",
          type: "password",
          placeholder: "Enter your password",
          classname: {
            container: "col-span-2",
            inputBasic: {
              input: "bg-[transparent] text-white",
              inputContainer: "border-2 p-[10px] rounded-md",
            },
          },
        },
      ];

    default:
      return [
        {
          label: "Email",
          name: "username",
          placeholder: "Enter your email",
          type: "email",
          classname: {
            container: "col-span-2",
            inputBasic: {
              input: "bg-[transparent] text-white",
              inputContainer: "border-2 p-[10px] rounded-md",
            },
          },
        },
        {
          label: "password",
          name: "password",
          type: "password",
          placeholder: "Enter your username",
          classname: {
            container: "col-span-2",
            inputBasic: {
              input: "bg-[transparent] text-white",
              inputContainer: "border-2 p-[10px] rounded-md",
            },
          },
        },
      ];
  }
};
