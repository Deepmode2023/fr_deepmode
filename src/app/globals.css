@tailwind base;
@tailwind utilities;
@tailwind components;

.min-h-100vh {
  min-height: 100vh;
}
/*
html.dark .bg-global {
  background-color: #221814;
}

html.dark .color-global {
  color: #221814;
}

html.dark .stroke-global {
  stroke: #221814;
}

html.light .bg-global {
  background-color: white;
  stroke: black;
} */
