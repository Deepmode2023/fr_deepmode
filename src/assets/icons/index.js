import { InfoSvg } from "./InfoSvg";
import { VerifySvg } from "./VerifySvg";
import { WarningSvg } from "./WarningSvg";
import { DangerSvg } from "./DangerSvg";
import { InfoCircleSvg } from "./InfoCircleSvg";
import { CloseSvg } from "./CloseSvg";
import { SettingsSvg } from "./SettingsSvg";
import { CheckSvg } from "./CheckSvg";
import { EyeSvg } from "./EyeSvg";
import { CloseEyeSvg } from "./CloseEyeSvg";

export {
  WarningSvg,
  EyeSvg,
  VerifySvg,
  InfoSvg,
  DangerSvg,
  InfoCircleSvg,
  CloseSvg,
  SettingsSvg,
  CheckSvg,
  CloseEyeSvg,
};
