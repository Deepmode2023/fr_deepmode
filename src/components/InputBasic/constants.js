export const SIZE_ICONS = 17;
export const COLORS = {
  success: "#42b883",
  usually: "#6B7280",
  error: "#f73963",
};

export const ID_TESTING = {
  input: "input-basic-input",
  placeholder: "input-basic-placeholder",
  icon: "input-basic-icon",
  label: "input-basic-label",
};
