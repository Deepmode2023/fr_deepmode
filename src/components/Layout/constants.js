export const SIZED_LAYOUT = {
  MAIN: { gap: 4 },
  NAVBAR: { minWidth: 100, maxWidth: 300 },
  AUTH_NAVBAR: { height: 100 },
};

export const LAYOUT_GRID = {
  DESC: {
    main: `grid-cols-[auto_1fr_auto_100px] grid-rows-[${
      SIZED_LAYOUT.AUTH_NAVBAR.height / 2
    }px_${
      SIZED_LAYOUT.AUTH_NAVBAR.height / 2
    }px_minmax(100px,300px)_auto] box-border gap-1`,
    // NAVBAR
    navbar: `col-start-1 col-end-1 row-start-1 row-end-5 min-w-[${SIZED_LAYOUT.NAVBAR.minWidth}px] 
                max-w-[${SIZED_LAYOUT.NAVBAR.maxWidth}px] relative box-border`,
    // AUTH NAVBAR
    authNavbar:
      "col-start-3 col-end-5  row-start-1 row-end-3 sticky top-0 z-50 relative box-border",
    // CONTENT
    content:
      "col-start-2 col-end-5 row-start-3 row-end-5 z-0 box-border relative",
    // NAVBAR TOOLTIP
    navbarTooltip:
      "col-start-2 col-end-5 row-start-3 row-end-4 z-60 box-border sticky top-0",
  },
};
