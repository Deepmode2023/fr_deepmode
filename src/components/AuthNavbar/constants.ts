import React from "react";
import { IAuthSubLinkProperty } from "@/interfaces/components/navbar/authnavbar";
import { SettingsSvg, AccountSvg } from "@/assets/icons";

export const AuthSubLinkProperty: Array<IAuthSubLinkProperty> = [
  {
    header: "Account",
    keyName: "account",
    iconHeader: AccountSvg,
  },
  {
    header: "Settings",
    keyName: "settings",
    iconHeader: SettingsSvg,
  },
];
