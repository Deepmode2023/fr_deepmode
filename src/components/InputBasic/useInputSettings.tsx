import { useEffect, useState } from "react";
import { StandartValidate } from "@/utils/validateInput";
import {
  SPECIAL_VALIDATE,
  IStandartValidateValue,
} from "@/interfaces/validate";
import { OnClickIconParamsType } from "./interface";
import { EyeSvg, CloseEyeSvg } from "@/assets/icons";
import { COLORS, SIZE_ICONS } from "./constants";

export type UseValueType = {
  onValidateInput?: (
    value: string,
    type: SPECIAL_VALIDATE
  ) => IStandartValidateValue;
  type: SPECIAL_VALIDATE;
  onClickIcon?: (params: OnClickIconParamsType) => void;
  icon?: (color: string) => JSX.Element;
};

export const useInputSettings = ({
  onValidateInput,
  type,
  onClickIcon,
  icon,
}: UseValueType) => {
  const [value, setValue] = useState("");
  const [touch, setTouch] = useState(false);
  const [visiblePassword, setVisisblePassword] = useState(false);
  const [typeInput, setTypeInput] = useState(type);
  const [iconButton, setIconButton] = useState<JSX.Element | null>(null);

  const conditionInput = onValidateInput
    ? onValidateInput(value, type)
    : StandartValidate(value, type);

  const color = !touch
    ? COLORS.usually
    : conditionInput.isValidate
    ? COLORS.success
    : COLORS.error;

  useEffect(() => {
    if (icon) setIconButton(icon(color));
    if (type === "password") {
      setIconButton(
        visiblePassword ? (
          <EyeSvg width={SIZE_ICONS} height={SIZE_ICONS} fill={color} />
        ) : (
          <CloseEyeSvg width={SIZE_ICONS} height={SIZE_ICONS} fill={color} />
        )
      );
    }
  }, [color, icon, type, visiblePassword]);

  const callbackIconClick = (params: OnClickIconParamsType): void => {
    setVisisblePassword(!visiblePassword);
    if (onClickIcon) return onClickIcon({ ...params, type });
    if (type === "password") {
      setTypeInput(!visiblePassword ? "text" : "password");
    }
  };

  useEffect(() => {
    if (value.length > 0) setTouch(true);
    else setTouch(false);
  }, [value]);

  return {
    value,
    setValue,
    touch,
    setTouch,
    iconButton,
    color,
    typeInput,
    callbackIconClick,
    isValid: conditionInput.isValidate,
    message: conditionInput.message,
  };
};
