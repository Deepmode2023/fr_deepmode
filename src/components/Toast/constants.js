import { VerifySvg, WarningSvg, DangerSvg } from "@/assets/icons";

export const HEIGHT_TOAST = 100;
export const WIDTH_TOAST = 400;
export const GAP_TOAST = 20;

export const calculate_postition_top = (position) => {
  if (position == 0) return GAP_TOAST;

  return position * HEIGHT_TOAST + GAP_TOAST * (position + 1);
};

export const COLORS_CONDITION = {
  success: { color: "#42b883", hover: "#379B6E", light: "#87F3C2" },
  error: { color: "#f73963", hover: "#a72b3b", light: "#fdecee" },
  warning: { color: "#205dd8", hover: "#2051B4", light: "#6999FB" },
};

export const ICONS = {
  success: {
    icon: (
      <VerifySvg
        width={30}
        height={30}
        strokeColor={COLORS_CONDITION.success.color}
      />
    ),
    title: "Great success!",
  },
  error: {
    icon: (
      <DangerSvg
        width={30}
        height={30}
        strokeColor={COLORS_CONDITION.error.color}
      />
    ),
    title: "Yikes. Something went wrong!",
  },
  warning: {
    icon: (
      <WarningSvg
        width={30}
        height={30}
        strokeColor={COLORS_CONDITION.warning.color}
      />
    ),
    title: "Check your input!",
  },
};
