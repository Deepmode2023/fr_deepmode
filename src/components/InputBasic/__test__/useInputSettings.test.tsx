import { renderHook, RenderHookResult, waitFor } from "@testing-library/react";
import {
  IStandartValidateValue,
  SPECIAL_VALIDATE,
} from "@/interfaces/validate";
import { expect, jest } from "@jest/globals";
import { StandartValidate } from "@/utils/validateInput";
import { useInputSettings, UseValueType } from "../useInputSettings";
import { EyeSvg, CloseEyeSvg } from "@/assets/icons";
import { COLORS, SIZE_ICONS } from "../constants";

Object.defineProperty(window, "scrollTo", { value: jest.fn() });

describe("Check correct work hooks useInputSettings", () => {
  const onValidInputMock = jest.fn(
    (value: string, type: SPECIAL_VALIDATE): IStandartValidateValue => ({
      isValidate: true,
      message: "some",
    })
  );
  const valueTyping = "Some value typing...";
  let hookInstance: RenderHookResult<
    ReturnType<typeof useInputSettings>,
    UseValueType
  >;

  const params: UseValueType = {
    type: "email",
    onValidateInput: onValidInputMock,
  };

  beforeEach(async () => {
    hookInstance = renderHook((props) =>
      useInputSettings({ ...(props ?? params) })
    );

    await waitFor(() => {
      hookInstance.result.current.setValue(valueTyping);
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("Check how the value parameter is changed", async () => {
    expect(hookInstance.result.current.value).toBe(valueTyping);
  });

  test("Checking how the touch parameter changes", async () => {
    expect(hookInstance.result.current.touch).toBe(true);

    await waitFor(() => {
      hookInstance.result.current.setValue("");
    });

    expect(hookInstance.result.current.touch).toBe(false);
  });

  test("Checking the validatetion function we passed in", async () => {
    //We have strick mode and 2 render after changes always
    expect(onValidInputMock.mock.calls).toHaveLength(3);
    expect(onValidInputMock).toHaveBeenCalledWith(valueTyping, params.type);
    expect(onValidInputMock.mock.results[1].value).toEqual({
      isValidate: true,
      message: "some",
    });
  });

  test("Checking the correct operation of the StandardValidate function", async () => {
    jest.clearAllMocks();

    const valueInsideHook = hookInstance.result.current.value;
    await waitFor(() => {
      hookInstance.rerender({ ...params, onValidateInput: undefined });
    });
    expect(valueInsideHook).toBe(valueTyping);
    expect(onValidInputMock).not.toHaveBeenCalled();
    expect(hookInstance.result.current.message).toBe(
      StandartValidate(valueInsideHook, params.type).message
    );
  });

  test("Checking the correct color of the display icon with the password type", async () => {
    await waitFor(() => {
      hookInstance.rerender({ onValidateInput: undefined, type: "password" });
    });

    expect(hookInstance.result.current.isValid).toBe(false);
    expect(hookInstance.result.current.iconButton).toEqual(
      <CloseEyeSvg fill={COLORS.error} height={SIZE_ICONS} width={SIZE_ICONS} />
    );

    await waitFor(() => {
      hookInstance.result.current.setValue("Test valid value1");
    });

    expect(hookInstance.result.current.isValid).toBe(true);
    expect(hookInstance.result.current.iconButton).toEqual(
      <CloseEyeSvg
        fill={COLORS.success}
        height={SIZE_ICONS}
        width={SIZE_ICONS}
      />
    );
  });

  test("Checking that the icon is displayed correctly with type password", async () => {
    await waitFor(() => {
      hookInstance.rerender({ onValidateInput: undefined, type: "password" });
    });
    const color = hookInstance.result.current.isValid
      ? COLORS.success
      : COLORS.error;

    expect(hookInstance.result.current.iconButton).toEqual(
      <CloseEyeSvg fill={color} width={SIZE_ICONS} height={SIZE_ICONS} />
    );

    await waitFor(() => {
      hookInstance.result.current.callbackIconClick({});
    });

    expect(hookInstance.result.current.iconButton).toEqual(
      <EyeSvg fill={color} width={SIZE_ICONS} height={SIZE_ICONS} />
    );
  });

  test("useInputSettings snapshot", () => {
    expect(hookInstance).toMatchSnapshot();
  });
});
