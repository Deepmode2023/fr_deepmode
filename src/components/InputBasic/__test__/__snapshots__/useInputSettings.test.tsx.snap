// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check correct work hooks useInputSettings useInputSettings snapshot 1`] = `
{
  "rerender": [Function],
  "result": {
    "current": {
      "callbackIconClick": [Function],
      "color": "#42b883",
      "iconButton": null,
      "isValid": true,
      "message": "some",
      "setTouch": [Function],
      "setValue": [Function],
      "touch": true,
      "typeInput": "email",
      "value": "Some value typing...",
    },
  },
  "unmount": [Function],
}
`;
